OMEdit/OMEdit/OMEditGUI/Annotations/ShapePropertiesDialog.cpp

```diff
@@ -201,10 +201,28 @@ ShapePropertiesDialog::ShapePropertiesDialog(ShapeAnnotation *pShapeAnnotation,
     mpFontNameComboBox->setCurrentIndex(0);
   }
   mpFontSizeLabel = new Label(Helper::size);
-  mpFontSizeSpinBox = new DoubleSpinBox;
-  mpFontSizeSpinBox->setRange(0, std::numeric_limits<double>::max());
-  mpFontSizeSpinBox->setValue(mpShapeAnnotation->getFontSize());
-  mpFontSizeSpinBox->setSingleStep(1);
+  mpFontSizeComboBox = new QComboBox;
+  mpFontSizeComboBox->setEditable(true);
+  // get the standard font sizes
+  QList<int> standardFontSizes = QFontDatabase::standardSizes();
+  // remove the font sizes less than Helper::minimumTextFontSize
+  QMutableListIterator<int> mutableListIterator(standardFontSizes);
+  while (mutableListIterator.hasNext()) {
+    if (mutableListIterator.next() < Helper::minimumTextFontSize) {
+      mutableListIterator.remove();
+    }
+  }
+  // add the font sizes to the combobox
+  mpFontSizeComboBox->addItem("(Auto)");
+  for (int i = 0; i < standardFontSizes.size(); ++i) {
+    mpFontSizeComboBox->addItem(QString::number(standardFontSizes.at(i)));
+  }
+  // if the font size is less than 0 then set it to Auto
```

Comment:
should read less than or equal to 👍

Reply:
I think that will result in a index out of bound crash. QList is not MetaModelica like list.

Reply:
Ah, but the code below doe check  if (something <=  0), I just wanted the comment to reflect the code